### PR TITLE
torch.utils.data.Dataloader: documentation about RNG state consumption

### PR DESCRIPTION
**Pull request by InnovArul**
the outcome from the pytorch forum issue: https://discuss.pytorch.org/t/dataloader-problem-problem-arises-when-shuffle-true/45631

The discussion is here: https://github.com/pytorch/pytorch/pull/20749

### CODE REVIEW

File: torch/utils/data/dataloader.py [hunk 1]
```diff
@@ -101,6 +103,9 @@ class DataLoader(object):
                  :ref:`multiprocessing-best-practices` on more details related
                  to multiprocessing in PyTorch.
 
+    .. note:: By default, ``DataLoader`` consumes a RNG state to determine ``base_seed`` 
+              for data-loading threads.
```
Comment (SsnL) [inline torch/utils/data/dataloader.py]: Can you instead add the note to https://github.com/pytorch/pytorch/blame/4ba1c4f798fa0aeb00705d8f545168d183393a38/docs/source/data.rst#L336?

Also, the workers are separate processes, not threads.

Comment (InnovArul) [reply torch/utils/data/dataloader.py]: I have moved the note to `data.rst`. Apparently, it was already there in `data.rst` that RNG was used to generate `base_seed`. Just added a side note to it to say its mandatory. Pl let me know if there is any change needed.

**Review by ssnl — APPROVED** [review]
Thank you :)
